modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java

```diff
@@ -350,11 +350,11 @@ protected void parseCreateField(ParseContext context) throws IOException {
     }
 
     @Override
-    protected Object parseSourceValue(Object value, String format) {
+    public ValueFetcher valueFetcher(SearchLookup lookup, String format) {
         if (format != null) {
             throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
         }
-        return value;
+        return sourceValueFetcher(lookup, v -> v);
```

Comment:
I'd prefer `Function.identity()` but this has to be a `CheckedFunction` to handle the `IOException` that reading doc values can throw. Runtime fields might throw an `IOException` because they read from doc values.